### ISSUE TITLE
Make memory a custom class, with an overloaded index operator for bounds checking.

### ISSUE THREAD

**Issue by bastisawesome** [bastisawesome/BZ80#12]
When going over the rewrite, originally I was going to do this to solve an issue with ROMs, but then I found out ROM isn't read independently of RAM, and opted to not do this. Thankfully, I found this TODO in the code.

There is one major problem with this issue, however: The Z80 is capable of reading up to 4 independent RAM, ROM, and PROM, so defining a memory system that can support this (as is intended for this emulator) will be much more complicated. This issue is solely to resolve the title problem, however, and a separate issue exists to track progress on building a more accurate memory system that can handle multiple ROMs or RAMs.

In short: this issue will be resolved solely with a memory class that will ensure bounds checking is done properly, despite `std::array` including a method that can do bounds checking.
